com.unity.hlod/Editor/HLODMeshSetterEditor.cs

```diff
@@ -0,0 +1,195 @@
+    public class HLODMeshSetterEditor : Editor
+    {
+        class Styles
+        {
+            public const float kSingleLineHeight = 18.0f;
+            public const float kSpacing = 5;
+            public readonly GUIContent meshSettingHeader = new GUIContent("Mesh settings");
+            public readonly GUIContent meshRenderersHeader = new GUIContent("Mesh renderers");
+        }
+
+        class MeshGroupUI
+        {
+            private SerializedProperty m_property;
+            private ReorderableList m_meshRendererList;
+
+            private SerializedProperty m_targetLevelProperty;
+            private SerializedProperty m_meshRenderersProperty;
+            public MeshGroupUI(SerializedObject serializedObject, SerializedProperty serializedProperty)
+            {
+                m_property = serializedProperty;
+
+                m_targetLevelProperty = m_property.FindPropertyRelative("m_targetLevel");
+                m_meshRenderersProperty = m_property.FindPropertyRelative("m_meshRenderers");
+
+                m_meshRendererList =
+                    new ReorderableList(serializedObject, m_meshRenderersProperty, false, true, true, true);
+                
+                m_meshRendererList.drawHeaderCallback += DrawHeaderCallback;
+                m_meshRendererList.drawElementCallback += DrawElementCallback;
+                m_meshRendererList.elementHeightCallback += ElementHeightCallback;
+                m_meshRendererList.onAddCallback += OnAddCallback;
+                m_meshRendererList.onRemoveCallback += OnRemoveCallback;
+            }
+            private void DrawHeaderCallback(Rect rect)
+            {
+                EditorGUI.LabelField(rect, styles.meshRenderersHeader);
+            }
+            private void DrawElementCallback(Rect rect, int index, bool isactive, bool isfocused)
+            {
+                Rect objectRect = new Rect(rect.x, rect.y, rect.width * 0.6f, rect.height);
+                Rect labelRect = new Rect(rect.x + objectRect.width + 10, rect.y, rect.width - objectRect.width, rect.height);
+
+                var meshRendererProp = m_meshRenderersProperty.GetArrayElementAtIndex(index);
+                EditorGUI.ObjectField(objectRect, meshRendererProp, typeof(MeshRenderer), GUIContent.none);
+
+                int triCount = 0;
+                int meshCount = 0;
+                var meshRenderer = meshRendererProp.objectReferenceValue as MeshRenderer;
+
+                if (meshRenderer != null)
+                {
+                    var meshFilter =meshRenderer.GetComponent<MeshFilter>();
+                    if (meshFilter != null)
+                    {
+                        //Actually, GetTriangleCount is return the index count.
+                        //in the mesh, Index is called Triangle.
+                        //So after divided by 3, we can get a real triangle count.
+                        triCount = meshFilter.sharedMesh.GetTriangleCount() / 3;
```

Comment:
The implementation of GetTriangleCount() doesn't seem to consider the topology type. This may not be a big issue, but in cases with other topology types except MeshTopology.Triangles the function may cause subtle unexpected troubles. Thus, I would like to suggest using Mesh.triangles instead as when I looked through the property's implementation it always generates triangle indices unlike Mesh.GetIndexCount(). Another option can be GetTriangles() with that we can get the triangle count of a sub mesh separately.

Reply:
Changed to mesh.triangles.Length.